gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -140,9 +140,13 @@ public FileAwareInputStreamDataWriter(State state, int numBranches, int branchId
     this.fs = FileSystem.get(uri, conf);
     this.fileContext = FileContext.getFileContext(uri, conf);
 
-    this.stagingDir = this.writerAttemptIdOptional.isPresent() ? WriterUtils
-        .getWriterStagingDir(state, numBranches, branchId, this.writerAttemptIdOptional.get())
-        : WriterUtils.getWriterStagingDir(state, numBranches, branchId);
+    if (state.getPropAsBoolean(ConfigurationKeys.USER_DEFINED_STAGING_DIR_FLAG)) {
```

Comment:
Think about maintaining the backward-compatibility for such kind of change.

Reply:
For backward-compatibility, I have added default value of USER_DEFINED_STAGING_DIR_FLAG as false, also, in disctp template the value for the above flag is false